### ISSUE TITLE
Add Support for Notifications on New Changelog Updates

### ISSUE THREAD

**Issue by SudharakaP** [opencollective/opencollective#4345]
We need to provide users with notifications when a new change-log update has been submitted. The design for this is explained in, 🐨 

https://user-images.githubusercontent.com/12435965/121241835-ba17e880-c850-11eb-86f4-1acfe501823f.mp4

For the initial MPV here are some of my thoughts, 🧠 🐨 

- For each user we will need to track whether they have seen new updates or not. Thus for whenever a new update is published we should have a flag for each user to track whether they've seen the updates or not (maybe in `collectives` table for role type `USER`). 🎏 

- Once the user clicks on the yellow banner or (fire icon) and the modal is shown to the user we can set the flag so that the user will not be shown the notification again. 🐨 

Figma Designs: https://www.figma.com/file/ZQBMWhnGGtRWeIZknFW1eP/%5BOC-Design%5D-Production-Ready-%E2%9C%85?node-id=12788%3A152534

cc: @Betree

**Comment by Betree** [comment]
It would be great to have something more specific for the specs, especially on where and how are we going to store this info about seen updates (a specific column? collective settings?). I would personally advocate for a dedicated table to prevent overloading Collectives yet again and to provide better indexing performances since we're going to fetch this info quite often (at least on each login).

Also, what will the GraphQL API look like? We should define the basic fields, mutations, and queries that will be needed to implement the feature.

**Comment by SudharakaP** [comment]
> It would be great to have something more specific for the specs, especially on where and how are we going to store this info about seen updates (a specific column? collective settings?). I would personally advocate for a dedicated table to prevent overloading Collectives yet again and to provide better indexing performances since we're going to fetch this info quite often (at least on each login).

@Betree : I was talking about this with @znarf as well today and we have several approaches I think. 

- A new table where we have the following structure; three columns, `id`, `userId`, `changelogViewDate`. So in this case when the user logs in we are checking if the `changelogViewDate` is earlier than the latest changelog publish date, and if so we show the notification. 🐨 

- The one I was thinking about is more along the lines of adding a column (say `changelogViewDate`) to the collectives table to and as soon as the user opens the notifications we we check if the `changelogViewDate` is earlier than the latest changelog publish date, and if so we show the notification. 🐨 

Note that in both of these approaches we are not tracking the whether and individual update has been seen by the user, only whether the user has opened the change log modal or not. I feel this is good enough for the first iteration. 🤔 WDYT? So basically if we push one changelog update and the then another one and by the time the user logs in there's two changelog updates. They see the notification and open the modal and whether they scroll to the next update or not, regardless we will not show the notification again. My personal opinion is that this is okay since we don't want to overload the user with too many notifications.

However maybe in the future we can add something to the design where we could show in the modal that there's more new updates; not just the first one that the user sees when the first open the modal. In this regard I am biased to the first approach as it would be easier to support this case when we have a separate table from the beginning. 🤔 

> Also, what will the GraphQL API look like? We should define the basic fields, mutations, and queries that will be needed to implement the feature.

So, fields;

```
type ChangeLogNotification {
  id: String
  user: Account!
  changelogViewDate: DateTime
}
```
Mutations; 

```
mutation UpdateChangeLogNotification($user: AccountReferenceInput!, $changelogViewDate: DateTime!) {
  createChangeLogNotification(user: $user, changelogViewDate: $changelogViewDate) {
    id
    user
    changelogViewDate
  }
}
```
Queries; 

```
changeLogNotification(
    id: String
    user: String
    throwIfMissing: Boolean = false
): ChangeLogNotification
```

Let me know your thoughts; hope I covered everything, but please let me know otherwise. 🤔 🙂

**Comment by Betree** [comment]
> Note that in both of these approaches we are not tracking the whether and individual update has been seen by the user

That's where we need to make a decision. I'm perfectly fine with adding a column (or a field in `user.data` if it's just a temporary solution) if the goal is to track the `lastChangelogViewDate`. However, according to the designs we saw, it sounds to me that we will also want to track this info for each update in the future. I had this intent in mind when I proposed to use a separate table, something like (`UserId`, `UpdateId`, `viewedAt`).

But it will be interesting to have both info anyway, so yes - if you're more comfortable with it let's start with adding a column at the user's level to track the last time when they opened the modal.

Regarding the GraphQL API:
- Maybe `changelogViewDate`: can simply be a field on the `User`/`Individual` objects?
- `createChangeLogNotification`: the name doesn't seem very clear to me. I can suggest `setChangelogViewDate`
